zebra-state/src/service/finalized_state/disk_format/upgrade.rs

```diff
@@ -82,21 +83,39 @@ pub trait DiskFormatUpgrade {
 
 fn format_upgrades(
     min_version: Option<Version>,
-) -> impl Iterator<Item = Box<dyn DiskFormatUpgrade>> {
+) -> impl DoubleEndedIterator<Item = Box<dyn DiskFormatUpgrade>> {
     let min_version = move || min_version.clone().unwrap_or(Version::new(0, 0, 0));
 
     // Note: Disk format upgrades must be run in order of database version.
     ([
         Box::new(prune_trees::PruneTrees),
         Box::new(add_subtrees::AddSubtrees),
         Box::new(tree_keys_and_caches_upgrade::FixTreeKeyTypeAndCacheGenesisRoots),
-        // Value balance upgrade
-        Box::new(no_migration::NoMigration::new(26, 0, 0)),
-    ] as [Box<dyn DiskFormatUpgrade>; 4])
+        Box::new(no_migration::NoMigration::new(
+            "add value balance upgrade",
+            Version::new(26, 0, 0),
+        )),
+        Box::new(block_info_and_address_received::Upgrade),
+    ] as [Box<dyn DiskFormatUpgrade>; 5])
         .into_iter()
         .filter(move |upgrade| upgrade.version() > min_version())
 }
 
+/// Returns a list of all the major db format versions that can restored from the
+/// previous major database format.
+pub fn restorable_db_versions() -> Vec<u64> {
+    format_upgrades(None)
+        .filter_map(|upgrade| {
+            let version = upgrade.version();
+            if version.minor == 0 && version.patch == 0 {
+                Some(version.major)
+            } else {
+                None
+            }
+        })
+        .collect()
+}
+
```

Comment:
Not sure about this, this indicates that every x.0.0 version is restorable, but I don't think this is necessarily true for < 26

Reply:
> I think is not correct - restorable_db_versions() is now returning just 27

Yeah, oops, thank you for noticing that. Fixed in https://github.com/ZcashFoundation/zebra/pull/9539/commits/bc03d0b05fa445cb0d2bf48730dc6f2edb7f1c37.

It may be better to add a boolean field to `NoMigration` indicating whether the new format version can reuse the previous major version, but Zebra can't currently reuse any previous major format except the one just before the current major format version anyway.